text/000-boost-stdlib-dev/text.md

```diff
@@ -0,0 +1,538 @@
+  SProp and primitive projections).
+  Stop including any other stdlib components as part of the `coq` opam
+  package and encourage maintainers of Coq packages in other distributions
+  to do the same.
+
+# Detailed design
+
+## Prelude
+
+We will keep a prelude shipped with Coq itself. Let's go with the
+content of the `theories/Init` directory of Coq to start with. This
+means that the scope of the standard library, adressed below, is
+everything but `Init` in the `theories` directory.
+
+## Packages
+
+### Proof of concept
+
+Here is a first proposal for packages, see this
+[proof of concept](https://github.com/proux01/roots) for details and small
+changes enabling these dependencies.
+
+### Dependency graph:
+
+![dependency graph](depends.png)
+
+### Details
+
+* coq-roots-logic
+  * The Coq standard library, basic logic
+  * Contains basic facts from Coq standard library in the namespace Coq.Logic
+    Look at coq-roots-classical-logic for more elaborate results in namespace
+    `Coq.Logic`.
+  * <details><summary>contains:</summary> Logic/Adjointification.v,
+    Logic/Eqdep_dec.v, Logic/JMeq.v, Logic/SetIsType.v,
+    Logic/Berardi.v, Logic/ExtensionalFunctionRepresentative.v,
+    Logic/ProofIrrelevance.v, Logic/StrictProp.v,
+    Logic/ConstructiveEpsilon.v, Logic/ExtensionalityFacts.v,
+    Logic/ProofIrrelevanceFacts.v, Logic/WeakFan.v, Logic/Decidable.v,
+    Logic/FunctionalExtensionality.v, Logic/PropExtensionalityFacts.v,
+    Logic/Eqdep.v, Logic/HLevels.v, Logic/PropFacts.v,
+    Logic/EqdepFacts.v, Logic/Hurkens.v,
+    Logic/RelationalChoice.v</details>
+* coq-roots-program
+  * The Coq standard library, program
+  * Support for dependently-typed programming.
+    This is required to use the `Program` command.
+    Namespace `Coq.Program`
+  * <details><summary>contains:</summary> Program/Syntax.v,
+    Program/Equality.v, Program/Subset.v, Program/Program.v,
+    Program/Basics.v, Program/Tactics.v, Program/Combinators.v,
+    Program/Wf.v, Program/Utils.v</details>
+* coq-roots-relations
+  * The Coq standard library, relations
+  * Relations (definitions and basic results).
+    Namespace `Coq.Relatons`
+  * <details><summary>contains:</summary>
+    Relations/Relation_Operators.v, Relations/Relation_Definitions.v,
+    Relations/Operators_Properties.v, Relations/Relations.v</details>
+* coq-roots-classes
+  * The Coq standard library, classes
+  * Most of namespace `Coq.Classes`
+  * <details><summary>contains:</summary>
+    Classes/Morphisms_Relations.v, Classes/RelationPairs.v,
+    Classes/Init.v, Classes/CMorphisms.v, Classes/SetoidTactics.v,
+    Classes/Equivalence.v, Classes/Morphisms.v, Classes/SetoidClass.v,
+    Classes/Morphisms_Prop.v, Classes/CEquivalence.v,
+    Classes/CRelationClasses.v, Classes/DecidableClass.v,
+    Classes/RelationClasses.v</details>
+* coq-roots-setoids
+  * The Coq standard library, setoids
+  * This is required to use the `setoid_rewrite` tactic.
```

Comment:
I'm not sure it makes sense to try to separate the supporting code for Program or setoid_rewrite from coq-core development

Reply:
That's indeed one of the points that should be discussed, no strong opinion on my side (as long as it remains out of the prelude). Note that with the dependencies, this amounts to a not totally negligible 48 files.

Reply:
In terms of loc, that's 7% of the current stdlib (14k lines out of 189k)

Reply:
Apparently, the impact can be limited to 12 files / 3483 lines, so let's do that.